backend/src/zac/core/services.py

```diff
@@ -839,85 +866,94 @@ def get_documenten(
         for iot in all_informatieobjecttypen
     }
 
-    documenten = factory(Document, found)
-
     # resolve relations
-    for document in documenten:
+    for document in found:
         document.informatieobjecttype = informatieobjecttypen[
             document.informatieobjecttype
         ]
 
-    # cache results
-    for document in documenten:
-        cache_key = f"document:{document.bronorganisatie}:{document.identificatie}"
-        cache.set(cache_key, document, timeout=AN_HOUR / 2)
-
-    return documenten, gone
+    return found, gone
 
 
-@cache_result(
-    "document:{bronorganisatie}:{identificatie}:{versie}", timeout=AN_HOUR / 2
-)
 def find_document(
     bronorganisatie: str, identificatie: str, versie: Optional[int] = None
 ) -> Document:
     """
     Find the document uniquely identified by bronorganisatie and identificatie.
+
     """
-    query = {"bronorganisatie": bronorganisatie, "identificatie": identificatie}
+    cache_key = "document:{bronorganisatie}:{identificatie}:{versie}"
```

Comment:
```suggestion
    cache_key = f"document:{bronorganisatie}:{identificatie}:{versie}"
```

and the fact the build didn't choke on this indicates that there's a missing test for the caching behaviour of this function :P